sled-agent/src/services.rs

```diff
@@ -3839,7 +3826,7 @@ impl ServiceManager {
         switch_zone_ip: Ipv6Addr,
         rack_network_config: &RackNetworkConfig,
     ) -> Result<(), Error> {
-        let log = &self.inner.log;
+        let log = &self.inner.log.new(o!("stage" => SWITCH_ZONE_INIT_STAGE));
```

Comment:
If it's possible, we may want to create a logger once and pass it as an argument to these switch-related functions.

Each time we call `log.new`, we're cloning a HashMap of all key-value pairs: https://docs.rs/slog/latest/src/slog/lib.rs.html#1221-1225

This isn't a deal-breaker, but it's a non-zero cost, so if we can "make it once + pass it around" that's preferable to creating a new logger for each sub-function